tools/mo/openvino/tools/mo/front/tf/CorrectPaddingsForPadAfterComplex.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2018-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from openvino.tools.mo.front.common.partial_infer.utils import int64_array
+from openvino.tools.mo.front.common.replacement import FrontReplacementSubgraph
+from openvino.tools.mo.front.subgraph_matcher import SubgraphMatch
+from openvino.tools.mo.front.tf.graph_utils import create_op_with_const_inputs
+from openvino.tools.mo.graph.graph import Graph
+from openvino.tools.mo.ops.concat import Concat
+
+
+class CorrectPaddingsForPadAfterComplex(FrontReplacementSubgraph):
+    """
+    There are TF models with the TF operation Complex that has two real tensors as arguments and returns the complex
+    tensor with real and imaginary parts given as arguments in port 0 and 1 respectively.
+
+    Although TF has a native support of complex numbers, OpenVINO doesn't have such support and emulates a complex
+    tensor with the shape [N_0, ..., N_{r - 1}] as a real tensor of the shape [N_0, ..., N_{r - 1}, 2] interpreting
+    any complex number as a tuple of the form
+        (real part, imaginary part)
+    That is, the emulated complex tensor has the rank r + 1, not r as in the TF model.
+
+    Hence, when we convert a subgraph of the form
+
+    Complex
+       |
+       |
+      Pad
+
+    we should correct pads_begin and pads_end adding zero at the end of pads_begin and pads_end.
+
+    The transformation performs such corrections.
+    """
+    enabled = True
+
+    def run_after(self):
+        from openvino.tools.mo.front.tf.pad_tf_to_pad import PadTFToPad
+        return [PadTFToPad]
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('complex', dict(op='Complex')),
+                ('pad', dict(op='Pad')),
+            ],
+            edges=[
+                ('complex', 'pad', {'in': 0}),
+            ])
```

Comment:
After `Pad` operation we can have other operations that we must adjust as well in some other models. For example, Gather operation attributes might need to be adjusted, etc.
In some other models it will be insufficient. Can we modify this pattern to be more strict for operations to guarantee the correct IR or do not convert the model at all?

Reply:
BTW, in which model we need this pattern because I haven't find it in Inpaint and Istanet?